.github/scripts/check-version.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+
+# Version checking script for Homebrew tap updates
+# Usage: check-version.sh <product_name> <incoming_version> <is_cask>
+# Outputs: should_update (true/false), current_version, incoming_version
```

Comment:
The lambda supported 3 modes: cask, formula, and [both](https://github.com/hashicorp/homebrew-tap/blob/master/util/lambda_trigger/lambda_trigger.go#L18-L19).  It would be helpful for future maintainers to know how to call the script for each mode.  In particular, for `both` should `is_cask` be set or not set?  Or would `both` mode not currently work correctly